src/node.rs

```diff
@@ -891,6 +905,38 @@ impl PbftNode {
         self._broadcast_message(&PbftMessageType::ViewChange, msg_bytes, state)
     }
 
+    /// Check the on-chain list of peers; if it has changed, update peers list and return true.
+    fn update_membership(&mut self, block_id: BlockId, state: &mut PbftState) -> bool {
+        // Get list of peers from settings
+        let settings = self
+            .service
+            .get_settings(
+                block_id,
+                vec![String::from("sawtooth.consensus.pbft.peers")],
+            ).expect("Failed to get settings");
+        let peers = get_peers_from_settings(&settings);
+        let new_peers_set: HashSet<PeerId> = HashSet::from_iter(peers.clone());
+
+        // Check if membership has changed
+        let old_peers_set: HashSet<PeerId> = HashSet::from_iter(state.peer_ids.iter().cloned());
+        let difference: HashSet<PeerId> = new_peers_set
+            .symmetric_difference(&old_peers_set)
+            .cloned()
+            .collect();
+
+        if !difference.is_empty() {
+            state.peer_ids = peers;
+            let f = ((state.peer_ids.len() - 1) / 3) as u64;
+            if f == 0 {
+                panic!("This network no longer contains enough nodes to be fault tolerant");
```

Comment:
Do we want to reject the block instead of panicking if it doesn't have enough peers for PBFT? It seems sane to just consider that a bad attempt at publishing a block.

Reply:
We can't reject the block; this is called from on_block_commit, so the block has already been committed.